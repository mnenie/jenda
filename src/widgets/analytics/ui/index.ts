import UsersChart from './UsersChart.vue';

export { UsersChart };
