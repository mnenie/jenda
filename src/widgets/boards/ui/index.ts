import TeamBoards from './TeamBoards.vue';
import ActivityСharts from './ActivityСharts.vue';

export { TeamBoards, ActivityСharts };
