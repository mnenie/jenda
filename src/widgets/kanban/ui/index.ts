import KanbanWrapper from './KanbanWrapper.vue';

export { KanbanWrapper };
