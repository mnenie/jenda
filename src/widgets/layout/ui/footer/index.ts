import FooterWelcome from './FooterWelcome.vue';

export { FooterWelcome };
