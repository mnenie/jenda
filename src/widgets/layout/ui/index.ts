export * from './sidebar';
export * from './header';
