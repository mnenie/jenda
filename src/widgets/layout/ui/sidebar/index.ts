import AppSidebar from './AppSidebar.vue';

export { AppSidebar };
