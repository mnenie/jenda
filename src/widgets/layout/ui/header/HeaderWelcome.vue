<script setup lang="ts">
import { computed, ref, shallowReactive } from 'vue';
import { useDark, useToggle } from '@vueuse/core';
import { useLanguage } from '@/shared/lib/composables';
import { useI18n } from 'vue-i18n';
import { UiButton, UiSelect } from '@/shared/ui';
import { redirect } from '@/shared/lib/helpers';
import { RouteNames } from '@/shared/config/consts';
import { Globe, Moon, Sun } from 'lucide-vue-next';
import type { Options } from '@/shared/ui/select/types';
import { headerLinks } from '../../config/headerNavs';
import type { HeaderNavLink } from '../../types';

const isDark = useDark();
const toggleDark = useToggle(isDark);

const { tm, t } = useI18n();

const iconUrl = computed(() => {
  return isDark.value ? '/icons/kanban-dark.png' : '/icons/kanban.png';
});

const languages = shallowReactive<Options[]>([
  { name: 'EN', value: 'en-US' },
  { name: 'RU', value: 'ru-RU' },
  { name: 'ZH', value: 'zh-CN' }
]);
const language = ref('');

useLanguage(languages, language);

const links = computed(() => {
  const localeArr = tm('welcome.header.links') as HeaderNavLink[];
  return headerLinks.map((link, index) => ({
    ...link,
    name: localeArr[index]
  }));
});
</script>

<template>
  <div :class="$style.header_welcome">
    <div :class="$style.left_container">
      <div :class="$style.box">
        <div :class="$style.inside">
          <div :class="$style.name_container" @click="redirect('https://github.com/mnenie/jenda')">
            <img :src="iconUrl" />
            <h3 class="text-xl">Jenda</h3>
          </div>
          <UiButton v-for="link in links" :key="link.id" variant="ghost" size="sm" :class="$style.item">
            {{ link.name }}
          </UiButton>
        </div>
      </div>
      <div :class="$style.additional">
        <UiSelect v-model="language" :options="languages" as="btn">
          <Globe :size="16" />
        </UiSelect>
        <div :class="$style.separator" />
        <UiButton
          v-tooltip="'Change theme'"
          variant="ghost"
          size="sm"
          :class="$style.btn"
          @click="toggleDark()"
        >
          <Moon v-if="!isDark" :size="17" />
          <Sun v-else :size="17" />
        </UiButton>
      </div>
    </div>
    <div :class="$style.btns">
      <UiButton variant="ghost" style="font-weight: 500" @click="$router.push({ name: RouteNames.login })">
        {{ t('welcome.header.login') }}
      </UiButton>
      <UiButton @click="$router.push({ name: RouteNames.registration })">
        {{ t('welcome.header.reg') }}
      </UiButton>
    </div>
  </div>
</template>

<style module lang="scss">
@import '@/app/styles/mixins';

.header_welcome {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1040;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 10px 16px;

  .name_container {
    display: flex;
    align-items: center;
    gap: 4px;
    cursor: pointer;
    margin-right: 8px;

    & > img {
      width: 22px;
      height: 22px;
    }
  }

  .btns {
    display: flex;
    align-items: center;
    gap: 6px;

    .ghost_exception {
      font-weight: 500;
    }
  }

  .inside {
    display: flex;
    align-items: center;
    gap: 2px;

    .item {
      color: var(--zinc-400);
      font-weight: 500;
      cursor: pointer;

      @include on-hover {
        color: var(--zinc-700);
      }
    }
  }

  .left_container {
    display: inline-flex;
    align-items: center;
    gap: 8px;

    .additional {
      display: flex;
      align-items: center;
      gap: 4px;

      .btn {
        padding: 8px;
        gap: 4px;
        color: var(--zinc-300);

        @include on-hover {
          color: var(--zinc-600);
        }
      }
    }
  }
}

.separator {
  width: 1px;
  height: 16px;
  background-color: var(--zinc-200);
}

.box {
  position: relative;
  z-index: 2;
  display: flex;
  align-items: center;
  padding: 0px 10px 0px 15px;
  height: 40px;
  border-radius: 8px;
  border: 0.5px solid #e3e4e8;
  background: #fff;
  box-shadow:
    0 12px 40px 0 rgba(0, 0, 0, 0.06),
    0 1px 0 0 rgba(0, 0, 0, 0.04);
}

:global(html.dark) {
  .box {
    background-color: rgba(var(--zinc-rgb-600), 0.4);
    border-color: var(--zinc-600);
    box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);
  }
  .separator {
    background-color: var(--zinc-600);
  }
  .header_welcome {
    .inside {
      .item {
        color: var(--zinc-300);
        @include on-hover {
          color: var(--zinc-200);
          background-color: var(--zinc-600);
        }
      }
    }
    .left_container {
      .additional {
        .btn {
          @include on-hover {
            color: var(--zinc-200);
          }
        }
      }
    }
  }
}
</style>
