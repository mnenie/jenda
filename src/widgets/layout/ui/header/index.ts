import HeaderMain from './HeaderMain.vue';
import HeaderWelcome from './HeaderWelcome.vue';

export { HeaderMain, HeaderWelcome };
