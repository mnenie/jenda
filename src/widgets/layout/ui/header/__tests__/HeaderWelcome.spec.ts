import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import { nextTick } from 'vue';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import HeaderWelcome from '../HeaderWelcome.vue';
import i18n from '@/shared/lib/i18n';
import { Moon, Sun } from 'lucide-vue-next';
import { UiButton, UiSelect } from '@/shared/ui';

const mockRouter = {
  push: vi.fn(),
  beforeEach: vi.fn()
};

describe('tests for HeaderWelcome.vue', () => {
  const wrapper = mount(HeaderWelcome, {
    global: {
      plugins: [i18n],

      mocks: {
        t: (key: string) => {
          const translations: Record<string, string> = {
            'welcome.header.login': 'Log In',
            'welcome.header.reg': 'Registration'
          };
          return translations[key];
        },
        tm: (key: string) => {
          const translationsArr: Record<string, string[]> = {
            'welcome.header.links': []
          };
          return translationsArr[key];
        },
        $router: mockRouter
      }
    }
  });

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render subcomponents', () => {
    expect(wrapper.findComponent(UiButton).exists()).toBe(true);
    expect(wrapper.findComponent(UiSelect).exists()).toBe(true);
  });

  it('should handle mode + icon changing', async () => {
    const darkModeButton = wrapper.find('.btn');
    //@ts-expect-error instance
    wrapper.vm.isDark = false;

    expect(wrapper.findComponent(Moon).exists()).toBe(true);
    expect(wrapper.findComponent(Sun).exists()).toBe(false);

    darkModeButton.trigger('click');
    //@ts-expect-error instance
    wrapper.vm.isDark = true;
    await nextTick();
    expect(wrapper.findComponent(Moon).exists()).toBe(false);
    expect(wrapper.findComponent(Sun).exists()).toBe(true);
  });

  it('should redirect correctly to login', async () => {
    const loginButton = wrapper.find('.btns').findAllComponents(UiButton).at(0);

    await loginButton.trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'login' });
  });

  it('should redirect correctly to registration', async () => {
    const registrationButton = wrapper.find('.btns').findAllComponents(UiButton).at(1);

    await registrationButton.trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'registration' });
  });
});
