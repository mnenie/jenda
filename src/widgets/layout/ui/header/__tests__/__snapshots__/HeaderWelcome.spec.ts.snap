// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for HeaderWelcome.vue > should be render correctly 1`] = `
"<header class="header_welcome">
  <section class="containers">
    <div class="left_container">
      <div class="box">
        <div class="inside">
          <div class="name_container"><img src="/icons/kanban.png">
            <h3 class="text-xl">Jenda</h3>
          </div><button class="text-sm button ghost sm item">About</button><button class="text-sm button ghost sm item">Kanban</button><button class="text-sm button ghost sm item">Members</button><button class="text-sm button ghost sm item">Templates</button><button class="text-sm button ghost sm item">Collaborative</button><button class="text-sm button ghost sm item">Chats</button>
        </div>
      </div>
      <div class="additional">
        <div class="btn_variant"><button class="text-sm button ghost sm btn"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-globe-icon">
              <circle cx="12" cy="12" r="10"></circle>
              <path d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"></path>
              <path d="M2 12h20"></path>
            </svg> EN</button>
          <transition-stub name="dropdown" appear="false" persisted="false" css="true">
            <!--v-if-->
          </transition-stub>
        </div>
        <div class="separator"></div><button class="text-sm button ghost sm btn"><svg xmlns="http://www.w3.org/2000/svg" width="17" height="17" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-moon-icon">
            <path d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"></path>
          </svg></button>
      </div>
    </div>
    <div class="btns"><button class="text-sm button ghost md" style="font-weight: 500;">Log In</button><button class="text-sm button default md">Registration</button></div>
  </section>
</header>"
`;
